services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/task.py

```diff
@@ -276,8 +277,31 @@ async def retrieve_service_inputs(
             port_keys=port_keys,
         )
 
+        if scheduler_data.restart_policy == RestartPolicy.ON_INPUTS_DOWNLOADED:
+            logger.info("Will restart containers")
+            await dynamic_sidecar_client.restart_containers(
+                scheduler_data.dynamic_sidecar.endpoint
+            )
+            logger.info("Containers restarted")
+
         return RetrieveDataOutEnveloped.from_transferred_bytes(transferred_bytes)
 
+    async def restart_containers(self, node_uuid: NodeID) -> None:
+        """Restarts containers without saving or restiring the state or I/O ports"""
```

Comment:
```suggestion
        """Restarts containers without saving or restoring the state or I/O ports"""
```

so indeed save/restore or not?